up-l3/udiscovery/v3/service.adoc

```diff
@@ -0,0 +1,250 @@
+----
+
+
+== Replication API
+
+
+=== SetServiceTopics()
+The `SetServiceTopics()` API is used by the uDiscovery business logic to add, update, or remove information recursively between the local, domain, and central instances. the API is passed `SetServiceTopicsRequest` containing a repeated list of `UServiceTopicInfo` and `ttl` for when the info will expire. the function returns `SetServiceTopicsResponse` message if the method was successfully processed or it returns `UStatus` containing the error values specified below. 
+
+[.specitem,oft-sid="dsn~discovery-replication-client~1",oft-needs="impl"]
+--
+* link:../languages.adoc[uProtocol Language Libraries] *MUST NOT* implement the client-side of xref:../../../up-core-api/uprotocol/core/udiscovery/v3/udiscovery_replicator.proto[udiscovery_replicator.proto] API as these APIs are only used internally by uDiscovery business logic.
+--
+
+To replicate new or updated `UServiceTopic` metadata:
+[.specitem,oft-sid="dsn~discovery-setservicetopic-update~1",oft-needs="impl,test"]
+--
+* `ttl` value in `SetServiceTopicsRequest` *MUST* be set to the duration the metadata is valid for or absent from the message to indicate the metadata is valid forever.
+--
+
+To replicate removal of `UServiceTopic` metadata:
+[.specitem,oft-sid="dsn~discovery-setservicetopic-remove~1",oft-needs="impl,test"]
+--
+* `ttl` value in `SetServiceTopicsRequest` *MUST* be set to 0.
+--
+
+`SetServiceTopics()` Failure Reasons:
+
+[.specitem,oft-sid="dsn~discovery-setservicetopic-error-invalid~1",oft-needs="impl,test"]
+--
+*MUST* return `UCode.INVALID_ARGUMENT` if the URI passed does not contain a valid ue_id or a FQURI.
+--
+  
+[.specitem,oft-sid="dsn~discovery-setservicetopic-error-permission~1",oft-needs="impl,test"]
+--
+* *MUST* return `UCode.PERMISSION_DENIED` if the caller is not uDiscovery service in the correct recursive order (i.e. local -> domain -> central).
+* *MAY* grant access to deployment specific uEs such as a the software manager that installs or removes uEs.
+--
+
+[.specitem,oft-sid="dsn~discovery-setservicetopic-error-notfound~1",oft-needs="impl,test"]
+--
+* *MUST* return `UCode.NOT_FOUND` if no topic matching the URI was found when the `UServiceTopic` is being removed (i.e. `ttl=0`).
+--
+
+
+
+.Replication API Design
+[#udiscovery-replication-design]
+[mermaid]
+ifdef::env-github[[source,mermaid]]
+----
+sequenceDiagram
+
+OTAClient ->> Local-UDiscovery: SetServiceTopics()
+Local-UDiscovery -->> OTAClient: SetServiceTopicsResponse
+
+Local-UDiscovery ->> Domain-UDiscovery: SetServiceTopics()
+Domain-UDiscovery -->> Local-UDiscovery: SetServiceTopicsResponse
+
+Domain-UDiscovery ->> Central-UDiscovery: SetServiceTopics()
+Central-UDiscovery -->> Domain-UDiscovery: SetServiceTopicsResponse
+----
+
+
+== Implementation Consideration
+
+It is possible the information in domain or central instances might become out of sync with what isa stored in the other instances. In order to rectify this situation:
```

Comment:
spelling
... might become out of sync with what [is] stored in the other instances
```suggestion
It is possible the information in domain or central instances might become out of sync with what isa stored in the other instances. In order to rectify this situation:
```